src/components/PopupWithForm.js

```diff
@@ -22,8 +22,12 @@ export default class PopupWithForm extends Popup {
       evt.preventDefault();
       const formData = this._getInputValues();
       this._handleFormSubmit(formData);
-      this._form.reset();
-      super.close();
+      this.close();
     });
   }
-}
\ No newline at end of file
+
+  close() {
+    super.close();
+    this._form.reset();
```

Comment:
You should ** clear ** the inputs only after **submitting** a card rather than when you open or close popups because if you unexpectedly close the popup, you will be upset if the inputs are empty when you open it once more, and you’ll have to type the data again.

You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` or when you close the modal,  because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.